### DISCUSSION TITLE
Validation is not being triggered for the input field inside modal

### DISCUSSION THREAD

**Discussion by nimeshmaharjan1** [react-hook-form/react-hook-form#9578] [Q&A] [answered]
The link to the sandbox: 
https://codesandbox.io/s/cool-https-f1ooh2?file=/src/App.js

The form is being submitted when the modal is closed through the cross button or clicked anywhere outside the modal even when the input field is set as required.

You first need to open the modal and close it and then submit.

Any help would be really appreciated.

**Comment by nimeshmaharjan1** [comment]
![image](https://user-images.githubusercontent.com/86953478/207285422-2d41a1c7-2445-4bbd-a394-825bf6d818d1.png)

**Comment by Moshyfawn** [comment, marked as answer]
The input you want validate inside the modal is an uncontrolled one _(meaning it's `value` and validation `requirements` come from it being in the DOM / uses `register` method)_. So, for it's validation to be triggered, it needs to be rendered, but your modal unmounts it's children when closed. From [the `react-bootstrap` docs on Modal component](https://react-bootstrap.github.io/components/modal/#overview):
> Modals are _unmounted_ when closed.

I believe, you can pass `unmountOnExit={false}` to your `Modal` component to achieve your desired behaviour

**Reply by nimeshmaharjan1** [reply]
Ohh ok but bootstrap modal doesnt have that api. Is there a work around for this?

**Reply by Moshyfawn** [reply]
> Ohh ok but bootstrap modal doesnt have that api. Is there a work around for this?

Indeed, I didn't find this prop in the current Bootstrap docs, but I saw some issue from 2016 in their repo with a question about keeping the `Modal` `children` mounted an that's the prop they've suggested and it works in the sandbox you've sent as repro
